core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -504,9 +542,18 @@ protected Builder buildAggregate(Aggregate e, Builder builder,
           SqlStdOperatorTable.ROLLUP.createCall(SqlParserPos.ZERO, groupKeys));
     default:
     case OTHER:
+      // Make sure that the group sets contains all bits.
+      final List<ImmutableBitSet> groupSets;
+      if (aggregate.getGroupSet()
+          .equals(ImmutableBitSet.union(aggregate.groupSets))) {
+        groupSets = aggregate.getGroupSets();
+      } else {
+        groupSets = new ArrayList<>(aggregate.getGroupSets());
+        groupSets.add(0, aggregate.getGroupSet());
```

Comment:
I've no idea how many group sets would appear here, however, it would probably be safer to pre-size the array as `groupSets.size + 1`, then `groupSets.add(aggregate.getGroupSet())`, then `groupSets.addAll(aggregate.getGroupSets())`.

I guess the difference is not significant, however, it would be safer and it would avoid questions during review

Reply:
Done. Thank you for your advice.